cpp/src/parquet/arrow/arrow_schema_test.cc

```diff
@@ -727,6 +780,60 @@ TEST_F(TestConvertParquetSchema, ParquetRepeatedNestedSchema) {
   ASSERT_NO_FATAL_FAILURE(CheckFlatSchema(arrow_schema));
 }
 
+TEST_F(TestConvertParquetSchema, IllegalParquetNestedSchema) {
```

Comment:
This case verifies that three-level list and map cannot be nested in a legacy two-level list.